### PR TITLE
Enable Unit tests on CoreCLR

### PR DESCRIPTION
**Pull request by BrennanConroy**
#1547 
DesignTimeHost tests aren't on CoreCLR since Assembly class is non-inheritable on CoreCLR and it is used heavily.

### CODE REVIEW

File: test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs [hunk 1]
```diff
@@ -25,13 +25,17 @@ public static string GetRootedPath(params string[] paths)
                 return root;
             }
 
-            return Path.Combine(root, paths.Aggregate(Path.Combine));
+            return Path.Combine(root, paths.Aggregate((path1, path2) => Path.Combine(path1, path2)));
         }
 
         private static bool IsWindows()
```
Comment (victorhurdugaci) [inline test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs]: We should get rid of this method and use `PlatformHelper` instead

File: test/Microsoft.Framework.PackageManager.FunctionalTests/project.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
         "Microsoft.Framework.FunctionalTestUtils": "1.0.0-*",
         "Microsoft.Framework.Runtime.Common": "1.0.0-*",
         "Microsoft.Framework.PackageManager": "1.0.0-*",
+        "Microsoft.Framework.PackageManager.Tests": "1.0.0-*",
```
Comment (victorhurdugaci) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Why is functional tests depending on the (unit) tests project?

File: test/Bootstrapper.FunctionalTests/project.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "dependencies": {
         "Microsoft.Framework.FunctionalTestUtils": "1.0.0-*",
-        "Microsoft.Framework.PackageManager.FunctionalTests": "1.0.0-*",
+        "Microsoft.Framework.PackageManager.Tests": "1.0.0-*",
```
Comment (victorhurdugaci) [inline test/Bootstrapper.FunctionalTests/project.json]: Same question as below. Why this dependency?

File: test/Microsoft.Framework.PackageManager.FunctionalTests/project.json (hunk 2 above)
Comment (BrennanConroy) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: We used to do it the other way; Unit tests depended on functional tests. They share some Utilities. But now that the unit tests have dnxcore50 as a target framework and the functional tests "don't", moving the utilities to the unit tests makes it all work.

Comment (victorhurdugaci) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Ugh... That's ugly. Those utilities should be in a test common project

Comment (BrennanConroy) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Probably

Comment (davidfowl) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Not probably :smile:

Comment (BrennanConroy) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: That could be taken either way D:

**Comment by davidfowl** [comment]
> DesignTimeHost tests aren't on CoreCLR since Assembly class is non-inheritable on CoreCLR and it is used heavily.

What?

**Comment by BrennanConroy** [comment]
Look at the TestAssembly class and try to make it compile for both CoreCLR and Desktop.

File: test/Microsoft.Framework.PackageManager.FunctionalTests/project.json (hunk 2 above)
Comment (BrennanConroy) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: So do you want a "test common project" or not?

**Comment by davidfowl** [comment]
Make sure travis keeps working when you make this change

**Comment by BrennanConroy** [comment]
@davidfowl 
Travis passed!

File: test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs [hunk 4]
```diff
@@ -25,13 +25,7 @@ public static string GetRootedPath(params string[] paths)
...
             }
 
-            return Path.Combine(root, paths.Aggregate(Path.Combine));
-        }
-
-        private static bool IsWindows()
-        {
-            var p = (int)Environment.OSVersion.Platform;
-            return (p != 4) && (p != 6) && (p != 128);
+            return Path.Combine(root, paths.Aggregate((path1, path2) => Path.Combine(path1, path2)));
```
Comment (davidfowl) [inline test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs]: So you made my beautiful 1 liner more verbose why? :smile:

Comment (BrennanConroy) [inline test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs]: Because CoreCLR only has Path.Combine(string []) and it needed Path.Combine(string, string)

Comment (davidfowl) [inline test/Microsoft.Framework.Runtime.Tests/PathHelpers.cs]: @ellismg Can we fix this :smile:

File: test/Microsoft.Framework.PackageManager.FunctionalTests/project.json (hunk 2 above)
Comment (davidfowl) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Yes, fix it please, or move the logic elsewhere.

Comment (victorhurdugaci) [inline test/Microsoft.Framework.PackageManager.FunctionalTests/project.json]: Yes, please

**Comment by BrennanConroy** [comment]
Made a common test project

**Comment by davidfowl** [comment]
:shipit:

**Pull request closed without merging**